cocos/asset/assets/texture-2d.ts

```diff
@@ -94,6 +96,36 @@ export class Texture2D extends SimpleTexture {
         return this._mipmaps;
     }
     set mipmaps (value) {
+        if (value[0].mipmapLevelDataSize && value[0].mipmapLevelDataSize.length > 0) {
+            compressedImageAsset.length = 0;
+            const mipmapLevelDataSize = value[0].mipmapLevelDataSize;
+            const data: Uint8Array = value[0].data as Uint8Array;
+
+            let byteOffset = 0;
+            for (let i = 0; i < mipmapLevelDataSize.length; i++) {
+                const srcView = new Uint8Array(data.buffer, byteOffset, mipmapLevelDataSize[i]);
+                const dstView = new Uint8Array(mipmapLevelDataSize[i]);
+                dstView.set(srcView);
+                compressedImageAsset[i] = new ImageAsset({
+                    _data: dstView,
+                    _compressed: true,
+                    width: value[0].width,
+                    height: value[0].height,
+                    format: value[0].format,
```

Comment:
How about define a local variable for `value[0]` to avoid array index?

Reply:
Because the length and width of each layer of the mipmap has to point to the length and width of the mip at layer 0